plumbum/machines/ssh_machine.py

```diff
@@ -32,6 +54,21 @@ def close(self):
         """Closes(terminates) the tunnel"""
         self._session.close()
 
+    @property
+    def lport(self):
+        """Tunneled port or socket on the local machine."""
+        return self._lport
+
+    @property
+    def dport(self):
+        """Tunneled port or socket on the remote machine."""
+        return self._dport
+
+    @property
+    def reverse(self):
```

Comment:
Would you mind adding some typing? I know we don't fully type yet (or have much at all), but it's nice on new API and I think this one would be better with a `-> bool` on it.

Reply:
I would love to see Plumbum get more typing.

I had held off on typing this since the types of these new `SshTunnel()` properties need to match the types for `SshMachine().tunnel`'s arguments, which are currently untyped. My thought was that these properties should not be typed until the `SshMachine().tunnel` args that they come from get typed. If you want I can add typing to both `SshMachine().tunnel` and these properties. If so, a couple questions:

Should e.g. ports be typed as `int | str | None` or as some `Port` type that Plumbum can reuse elsewhere (`plumbum.lib.ptyping.Port`?)? If some `TypeAlias` `Port = int | str | None` is added, should `lport` be `Port` or (redundantly) `str | Port` to hint that `lport` can be a socket string too, not just a port string?

Reply:
I'd be fine to have a `plumbum.typing` module with some good reusable types, though I think this one is simple enough (and actually helps in reading) to leave `Union[int, str]` (or `"int | str"`; sadly we are still Python 3.6 compatible).

Than can be a followup PR if you'd like to help with it. :) I've been using monkeytype to try to get a head start on typing, but I haven't had much time to work on it (and won't till after SciPy & Snowmass, so mostly out all this month).

Reply:
I would like to help with this, but I also won't have time for a while :/